### ISSUE TITLE
add warning to confounds description page

### ISSUE THREAD

**Issue by satra** [poldracklab/fmriprep#1049]
1. if i'm already using aroma cleaned data, it does not make sense to further regress the aroma regressors. 
2. also it may not make sense to use global signal regression and a/t compcor 
3. if i'm doing my own high pass filtering, using the cosine filters could interfere

since it's so easy to add the entire confounds.tsv to a feat or other analysis, it may be useful to add a warning that people should selectively add confounds relevant to their specific analysis.

**Comment by chrisfilo** [comment]
We should add this to documentation. Maybe in red.

**Comment by alexlicohen** [comment]
Per the AROMA paper, WM and CSF confounds were regressed after AROMA. How does one do this with fMRIprep? I'm presuming we'll have to extract new timecourses using WM and CSF masks from the AROMA residual nii.gz?

> We validated ICA-AROMA by applying it to rfMRI and stop-signal task fMRI data (SST-fMRI). We compared results obtained with ICA-AROMA to two commonly applied motion correction strategies: extensive nuisance regression including 24 RPs, and spike regression. All three preprocessing strategies were complemented by **standard nuisance regression including time-series for WM and CSF signal** as well as a linear trend to remove residual (non-motion related) structured noise. 

> Therefore, motion artifacts are preferably removed prior to temporal filtering. To accommodate this, we designed ICA-AROMA to be applied prior to temporal filtering. Moreover, applying secondary motion artifact removal prior to temporal filtering not only prevents ringing artifacts but additionally prevents removal of low-frequency motion-related signal variance which can aid in identification of motion artifacts. **Similarly, as signal from WM and CSF time- series might also contain variance related to motion artifacts, we applied ICA-AROMA prior to nuisance regression**. Additionally, we applied ICA-AROMA after spatial smoothing. Spatial smoothing will blur motion-related artifacts to neighboring voxels. Yet, as spatial smoothing will increase signal to noise ratio by reducing speckled noise, it will increase the ability of ICA to detect coherent signal patterns, including better separation of structured artifacts. Note that we did not include slice-timing correction in our preprocessing pipeline. Although slice-timing correction might interact with motion-induced artifacts, it is not expected to substantially alter the characteristics of the motion components and therefore impact the ICA-AROMA classification accuracy.

> It should be noted that ICA-AROMA is specifically designed and validated with respect to removal of motion-related artifacts. Accordingly, we advise to check your data for any remaining structured artifacts and remove residual structured noise after application of ICA-AROMA. **We implemented the removal of residual noise by means of WM and CSF regression. Note that to prevent the risk of removing signal of interest due to spatial smoothing or inaccurate WM and CSF masks, a conservative approach is advisable. Here, we extracted mean WM and CSF signals using masks obtained by multiplying a participant-specific tissue prior with an MNI152-derived tissue prior, both thresholded at a very conservative threshold of 95% tissue probability.**

**Comment by satra** [comment]
this may help: http://fmriprep.readthedocs.io/en/latest/outputs.html#confounds

but yes, you will need to separately select the useful confounds and regress these out.

**Comment by alexlicohen** [comment]
This issue/question though, is that the columns in confounds.tsv represent the timecourses extracted from the data PRIOR to AROMA. 

So should we 're-extract' timecourses after AROMA using the same ROIs from the AROMA residual?
per #817 and #959 

(trying to come up with an ad-hoc solution in the meantime to understand our data.)

**Comment by satra** [comment]
ah yes! this would also require transforming the csf/wm masks (*T1w_dtissue.nii.gz) to mni space, since aroma, at present, generates everything in that space. or transforming the mni output of icaaroma to individual space.

@oesteban - it may be useful to add post-aroma operations to generate additional confounds.

**Comment by effigies** [comment]
What all confounds should be recomputed? Global, WM and CSF signals would be easy. FD and motion parameters don't make much sense, as motion will not be re-computed. I'm not sure about DVARS, aCompCor or tCompCor.

**Comment by satra** [comment]
> Global, WM and CSF signals would be easy. FD and motion parameters don't make much sense, as motion will not be re-computed.
+1

> I'm not sure about DVARS, aCompCor or tCompCor.

i think those would all change post aroma.

**Comment by effigies** [comment]
I agree they'd change, but would they be sensible to use post-AROMA?

Also, what should we call the recomputed columns? `<confound>_variant-AROMA`? Or just `<confound>_AROMA`? Other ideas?

**Comment by chrisfilo** [comment]
@alexlicohen The paper only clearly states when the regression was performed, but not at which stage the signals were extracted. Would be good to clarify with authors: @maartenmennes in the pipeline from the ICA-AROMA paper did you extract WM and CSF signals prior or post non-aggressive denoising?

@effigies Why not stick to CamelCase to keep names consistent? `WhiteMatterPostAROMA`?

Finally I just thought about one more approach that @alexlicohen might want to try: orthogonalizing WM and CSF regressor against the noise ICA AROMA regressors already present in the confounds file.

As a side note I have an impression that we are splitting hairs here and there is no clear answer how much the order of signal extraction steps influences the result and which order is the best one (+1 to what @satra said).

**Comment by alexlicohen** [comment]
To me, this is largely trying to avoid the "I'm doing the analysis a second way to address the concerns of Reviewer 2" problem...

**Comment by effigies** [comment]
> @effigies Why not stick to CamelCase to keep names consistent? WhiteMatterPostAROMA?

I was thinking of it from a FitLins perspective, such that if people pass in `--variant=AROMA` (or whatever we're calling the variant), to adjust `CSF` to `CSV_variant-AROMA` automatically, but the exact indicator doesn't need to match that on the filename.

So anyway, whatever people decide on is fine with me. From the perspective of the functional derivatives spec, we should discuss this case and, settling on some spec-endorsed separator `<sep>` (e.g. "Post"), define `<confound><sep><method>` as a variant of `<confound>` calculated after `<method>` was applied to the original data. Then we should have a table of `<method>`s that might be used (I assume AROMA isn't the only one), and it would probably be worth noting which confounds make sense to calculate and apply post-`<method>`, so that downstream projects can have an expectation of what to look for.

**Comment by oesteban** [comment]
If I interpret correctly, we also need to apply ICA-AROMA on the original time-series without motion correction, is that correct? @alexlicohen @satra

**Comment by alexlicohen** [comment]
no, they ran MELODIC after motion correction, intensity normalization, and 6mm FWHM smoothing. (no slice timing, but I would not advocate getting rid of it in fmriprep of course).

ICA-AROMA takes the place of regressing out the motion parameters, be they 6, 9, 12, 24, or 72:
> We defined a 72RP model, including the standard 24RP model (6 standard RPs, their derivatives, and the quadratic terms of these 12 RPs), as well as a single time-point backward and a forward shifted version of the 24RP model to reflect possible non-linear effects.

**Comment by oesteban** [comment]
> they ran MELODIC after motion correction

Okay, cool. Otherwise we could feed AROMA directly with motion-UNcorrected images and then apply motion correction on the AROMA-denoised outputs

**Comment by maartenmennes** [comment]
Aloha, two points:

- yes, any further variable you want to regress out is taken from the data after AROMA denoising. Obviously, if you want to include WM, CSF regression together WITH the regression of the AROMA identified components, you would extract these regressors from the preprocessing step just before doing that regression. In that sense, we assigned priority to regressing out motion-related components before regressing out other noise. (which relates to the whole discussion about aggressive vs. non-aggressive denoising)

- the 6mm smoothing kernel is a bit unfortunately displayed in the paper. We should have left the exact kernel out of that figure. For the data in the manuscript (3.5mm isotropic), we used 6mm smoothing, which comes from ceil(1.5 * the voxelsize), which is what we typically use as our smoothing kernel. So for 3mm data, we'd use 5mm.

- third minor point: under the KISS principle I would advocate getting rid of slice timing correction (we never apply that in our preprocessing). But that is a whole other discussion... :-)
